internal/venice-common/src/main/java/com/linkedin/venice/helix/HelixReadOnlySchemaRepository.java

```diff
@@ -380,6 +399,15 @@ public Optional<SchemaEntry> getSupersetSchema(String storeName) {
         throw new VeniceNoStoreException(storeName);
       }
       Optional<Integer> supersetSchemaID = getSupersetSchemaID(storeName);
+      if (supersetSchemaID.isPresent()) {
+        if (schemaData.getValueSchema(supersetSchemaID.get()) == null) {
+          logger.info(
+              "Superset schema ID: {} for store: {} not in cache, will force refresh. ",
+              supersetSchemaID.get(),
+              storeName);
+          forceUpdateSchemaData(storeRepository.getStore(storeName), schemaData);
+        }
+      }
       return supersetSchemaID.map(schemaData::getValueSchema);
```

Comment:
Need to verify if forcing refresh actually got the schema or not. else error out.

Reply:
Sounds good, I throw an exception when this happens in the latest diff.